app/assets/javascripts/sync.coffee

```diff
@@ -30,15 +32,29 @@ $ = jQuery
 
   onConnectFailure: (error) -> #noop
 
-  connect: -> @adapter.connect()
+  connect: ->
+    @adapter.connect()
+
+    onConnectInterval = setInterval (=>
+      return unless @isConnected()
+      clearInterval(onConnectInterval)
+      @adapter.onConnect()
+    ), 250
 
   isConnected: -> @adapter.isConnected()
 
-  onReady: (callback) ->
-    if @isReady()
-      callback()
-    else
-      @readyQueue.push callback
+  onDebug: (message) ->
```

Comment:
I'm still in favor of removing the `onDebug` function completely, unless there's a reason it needs to be added.